rosbag2_transport/include/rosbag2_transport/player_progress_bar.hpp

```diff
@@ -0,0 +1,96 @@
+{
+public:
+  enum class PlayerStatus : char
+  {
+    BURST = 'B',
+    DELAYED = 'D',
+    PAUSED = 'P',
+    RUNNING = 'R',
+    STOPPED = 'S',
+  };
+
+  /// PlayerProgressBar constructor
+  /// \param output_stream Reference to output stream where progress bar and help information
+  /// will be printed out. Could be std::cout, std::cerr or std::ostringstream for tests.
+  /// \param starting_time Time stamp of the first message in the bag.
+  /// \param ending_time Time stamp of the last message in the bag.
+  /// \param progress_bar_update_rate The progress bar maximum update rate in times per second (Hz).
+  /// If update rate equal 0 the progress bar will be disabled and will not output any information.
+  /// If update rate less than 0 the progress bar will be updated for every update(..) and
+  /// update_with_limited_rate(..) calls.
+  /// \param progress_bar_separation_lines Number of separation lines to print in between the
+  /// playback output and the progress bar.
+  explicit PlayerProgressBar(
+    std::ostream & output_stream,
+    rcutils_time_point_value_t starting_time,
+    rcutils_time_point_value_t ending_time,
+    int32_t progress_bar_update_rate = 0,
+    uint32_t progress_bar_separation_lines = 3);
+
+  virtual ~PlayerProgressBar();
+
+  /// \brief Prints help string about player progress bar.
+  /// Expected to be printed once at the beginning.
+  void print_help_str() const;
+
+  /// \brief Updates progress bar with the specified timestamp and player status, taking into
+  /// account the update rate set by the user.
+  /// \note The function should be used for regular progress bar updates, for example
+  /// after the publishing the next message.
+  /// \warning This function is not thread safe and shall not be called concurrently from multiple
+  /// threads.
+  /// \param status The player status to be updated on progress bar.
+  /// \param timestamp Timestamp of the last published message.
+  void update_with_limited_rate(
+    const PlayerStatus & status,
+    const rcutils_time_point_value_t & timestamp);
+
+  /// \brief Updates progress bar with the specified player status, irrespective to the update rate
+  /// set by the user.
+  /// \note The function should be called for extraordinary progress bar updates, for example
+  /// when player changed its internal status or a log message is printed, and we want to 'redraw'
+  /// the progress bar.
+  /// \param status The player status to be updated on progress bar.
+  /// \param timestamp Timestamp of the last published message. If timestamp is less than 0 the
+  /// last published timestamp will be taken from the inner progress bar cache.
+  void update(const PlayerStatus & status, const rcutils_time_point_value_t & timestamp = -1);
+
+  /// \brief Getter function for current player status.
+  /// \return Current player status.
+  PlayerStatus get_player_status();
+
+private:
+  std::unique_ptr<PlayerProgressBarImpl> pimpl_;
```

Comment:
this generates the linage warning with windows. (the CI actually detected it as warning.)

- https://ci.ros2.org/job/ci_windows/23386/msbuild/
- https://ci.ros2.org/job/ci_windows/23387/msbuild/

The warning is happening because the `PlayerProgressBarImpl` class is not marked with the `ROSBAG2_TRANSPORT_PUBLIC` macro, which is required for classes used in DLL interfaces.

Reply:
@fujitatomoya 
- The warnings you are mentioning was discovered last week and raised in the separate issue https://github.com/ros2/rosbag2/issues/1926.  
- I've merged fix this morning in the PR https://github.com/ros2/rosbag2/pull/1927

Reply:
@MichaelOrlov 👍 okay, i did not know that thanks!